### PR TITLE
ROX-20168: Hide image cve snooze UI when 2.0 deferral is enabled

### PR DESCRIPTION
**Pull request by dvail**
## Description

The UI side of https://github.com/stackrox/stackrox/pull/8128. Hides the legacy global snooze buttons in VM 1.0 when UNIFIED_DEFERRALS is enabled.

Easier to review with "Hide whitespace" enabled.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

With the flag enabled.

Toolbar buttons for Image CVEs are not present. Table row actions for CVE snooze are not present:
image
image

Buttons and actions *are* still available for Node and Platform CVEs:
image
image
image

With the flag disabled, the buttons and actions are once again available for image CVEs:
image
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* master
  * **PR #8216** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/8216" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/8216?utm_source=stack-comment).

**Comment by vjwilson** [comment]
> @vjwilson @alwayshooin - Does this cover all the places where the legacy "global snooze" is visible in the VM 1.0 UI?

Yes, that is the only place (same component, whether root or sidebar)

File: ui/apps/platform/src/Containers/VulnMgmt/List/Cves/VulnMgmtListCves.js [hunk 1]
```diff
@@ -295,13 +296,18 @@ const VulnMgmtCves = ({
...
+    const isUnifiedDeferralEnabled = isFeatureFlagEnabled('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL');
+
     const [selectedCveIds, setSelectedCveIds] = useState([]);
     const [bulkActionCveIds, setBulkActionCveIds] = useState([]);
 
     const workflowState = useContext(workflowStateContext);
 
     const cveType = workflowState.getCurrentEntityType();
 
+    const shouldRenderGlobalSnooze = !isUnifiedDeferralEnabled || cveType !== entityTypes.IMAGE_CVE;
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/VulnMgmt/List/Cves/VulnMgmtListCves.js:309]: I just wanted to make sure I'm getting the logic right. I had to look at it a bit. So if `ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL` is disabled, then the first part of this conditional will be `true`. That means the second part wouldn't matter, right? So can we potentially have `ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL` as disabled and the CVE type is not an Image CVE, and it'll still render? Wouldn't you only want to show for Image CVEs or am I mistaken?

**Review by sachaudh — APPROVED** [review]
LGTM. Just a clarification/question for you

File: ui/apps/platform/src/Containers/VulnMgmt/List/Cves/VulnMgmtListCves.js (hunk 1 above)
Comment (dvail) [reply ui/apps/platform/src/Containers/VulnMgmt/List/Cves/VulnMgmtListCves.js:309]: The components that this conditional renders are shown for image, node, and platform CVEs. The logic I was going for was:
- If the UNIFIED_DEFERRALS flag is not set, always render the snooze UI
- If the flag _is_ set, only render the snooze UI when the entity is something other than "IMAGE_CVE"

I actually tried this out as `!(isUnifiedDeferralEnabled && cveType === entityTypes.IMAGE_CVE)` for readability, but switched back after looking at it. Do you think that reads better than what I have here?

**Pull request merged**